### DISCUSSION TITLE
Docs unclear on adding remark plugins(?)

### DISCUSSION THREAD

**Discussion by ZenaMel** [facebook/docusaurus#8743] [Q&A]
Usually, adding remark plugins is easy. I'm having trouble with adding this plugin however: https://github.com/wataru-chocola/remark-extended-table/tree/main/packages/remark-extended-table#remark-extended-table

I tried:

1. `yarn add remark-extended-table`
2. Adding the plugin to remarkPlugins in docusaurus config:

```js
async function createConfig() {
  const remarkExtendedTable = (await import('remark-extended-table')).default;
  
 ...
 
         remarkPlugins: [
          [remarkExtendedTable, {'colspanWithEmptyCell': true}],
        ]
...
```

This doesn't do anything to the documentation, I'm unsure why though.

The "usage" docs of the plugin show this:

```js
import { remarkExtendedTable, extendedTableHandlers } from 'remark-extended-table';

import { unified } from 'unified';
import remarkParse from 'remark-parse';
import remarkRehype from 'remark-rehype';
import rehypeStringify from 'rehype-stringify';
import remarkGfm from 'remark-gfm';

const md = `
| a | b |
|---|---|
| 1 | 2 |
| ^ | 3 |
`;

const process = (md: string) =>
  unified()
    .use(remarkParse)
    .use(remarkGfm)
    .use(remarkExtendedTable)
    .use(remarkRehype, null, {
      handlers: {
        // any other handlers
        ...extendedTableHandlers,
      }
    })
    .use(rehypeStringify)
    .process(md);
```

But I'm unsure on how to translate this into Docusaurus. The [docusaurus docs](https://docusaurus.io/docs/markdown-features/plugins) don't say anything about this besides adding a plugin the way I did at the start.

The plugin docs also mention that trying to disable `remarkGfm` could solve issues, but I'm sure docusaurus doesn't use that by default anyway(?).

**Comment by slorber** [comment]
As far as I know, MDX 1 has GFM hardcoded so you can't turn it off. 

I guess this will work better after we upgrade to MDX 2, which will allow you to register this plugin just before remark-gfm

**Comment by Airkro** [comment]
This problem bother me for a long time too, there is no way to add `handlers` to `remark-rehype` in the `docusaurus` project.

You can use [rehype-extended-table](https://github.com/nice-move/rehype-extended-table) to replace `remark-extended-table`.

**Reply by homotechsual** [reply]
Yeah we really would benefit from a method to add remark-rehype handlers in Docusaurus - it's a fairly common pattern (remark-definition-list also uses it)

**Reply by slorber** [reply]
Would it solve the problem if we exposed `siteConfig.markdown.remarkRehypeOptions`?

This has also been requested in https://github.com/facebook/docusaurus/issues/9502

**Reply by homotechsual** [reply]
Yeah, that would solve the issue :-)

**Reply by CrazyOrr** [reply]
Yes, we really need a way to config `remark-rehype`.

**Reply by slorber** [reply]
PR to track: https://github.com/facebook/docusaurus/pull/9674
